### ISSUE TITLE
[KFP] Several issues with RBAC and pipelines

### ISSUE THREAD

**Issue by phvalguima** [canonical/bundle-kubeflow#394]
I am deploying Kubeflow on top of Charmed Kubernetes with "RBAC,Node" options for RBAC.

However, I am having several issues with Kubeflow Pipelines and RBAC.
In this case, I am deploying Kubeflow with "admin" as its user namespace.

1) Although described on issue #393, Kubeflow seems to come with multiuser enabled or at least needed:
First error I see is this: https://pastebin.ubuntu.com/p/Rx9S6vzn2m/
That is caused by the fact that neither kfp-api nor kfp-ui come with "KUBEFLOW_USERID_HEADER" and "KUBEFLOW_USERID_PREFIX" set on "config.json":
```
$ kubectl edit -n kubeflow cm kfp-ui-config-config
apiVersion: v1
data:
  config.json: '{"spec": {"serviceAccountName": "kubeflow-pipelines-viewer"}, "KUBEFLOW_USERID_HEADER":
    "kubeflow-userid", "KUBEFLOW_USERID_PREFIX": ""}'
kind: ConfigMap
metadata:
. . . .
```
After some digging, I've found this described on [kustomization files](https://github.com/kubeflow/pipelines/blob/dec03067ca1f89f1ca23c7397830d60201448fa6/manifests/kustomize/base/installs/multi-user/api-service/deployment-patch.yaml). [There is an equivalent option for ui service as well](https://github.com/kubeflow/pipelines/blob/dec03067ca1f89f1ca23c7397830d60201448fa6/manifests/kustomize/base/installs/multi-user/pipelines-ui/deployment-patch.yaml)

2) Once (1) is fixed, kfp cannot create an Experiment because of: https://pastebin.ubuntu.com/p/qDsxN9M6xn/ [1]
and: https://pastebin.ubuntu.com/p/fVY3tttxVh/ [2]
To test this issue, I've set a quite wide clusterrole for both kfp-api and admin service accounts:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.kubernetes.io/name: kfp-api
  name: kubeflow-admin
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kubeflow-admin
subjects:
- kind: ServiceAccount
  name: kfp-api
  namespace: kubeflow
- kind: ServiceAccount
  name: admin
  namespace: admin
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app.kubernetes.io/name: kfp-api
  name: kubeflow-admin
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
```

It is important to see that, according to [2], RBAC names must be "kubeflow-admin". I am not sure why kfp-api service is looking for this specific name thou.

Once these issues were sorted out, I've managed to create an Experiment.

However, Runs were failing because service account ```pipeline-runner``` needs to exist on both namespaces. 
The issues with Runs can be found at: 
```
$ kubectl get workflows.argoproj.io -n admin
```

I could not resolve this last problem:
```
Events:
  Type     Reason             Age    From                 Message
  ----     ------             ----   ----                 -------
  Normal   WorkflowRunning    2m46s  workflow-controller  Workflow Running
  Warning  WorkflowNodeError  2m46s  workflow-controller  Error node conditional-execution-pipeline-with-exit-handler-hwf56.exit-handler-1.flip-coin-op: pods "conditional-execution-pipeline-with-exit-handler-hwf56-1090332789" is forbidden: error looking up service account admin/pipeline-runner: serviceaccount "pipeline-runner" not found
  Warning  WorkflowNodeError  2m46s  workflow-controller  Error node conditional-execution-pipeline-with-exit-handler-hwf56.exit-handler-1.flip-coin-op: task 'conditional-execution-pipeline-with-exit-handler-hwf56.exit-handler-1.flip-coin-op' errored: pods "conditional-execution-pipeline-with-exit-handler-hwf56-1090332789" is forbidden: error looking up service account admin/pipeline-runner: serviceaccount "pipeline-runner" not found
  Warning  WorkflowNodeError  2m46s  workflow-controller  Error node conditional-execution-pipeline-with-exit-handler-hwf56.exit-handler-1
  Warning  WorkflowNodeError  2m46s  workflow-controller  Error node conditional-execution-pipeline-with-exit-handler-hwf56
  Warning  WorkflowNodeError  2m46s  workflow-controller  Error node conditional-execution-pipeline-with-exit-handler-hwf56.onExit: pods "conditional-execution-pipeline-with-exit-handler-hwf56-2645859952" is forbidden: error looking up service account admin/pipeline-runner: serviceaccount "pipeline-runner" not found
```

Even with the creation of "pipeline-runner" on the admin namespace (which means a 2nd service account named pipeline-runner), the Runs were not getting rendered correctly on kfp-ui.

**Comment by phvalguima** [comment]
If I disable RBAC, I can get a pipeline deployed with:

``` 
$ kubectl create sa -n admin pipeline-runner
$ kubectl get secret mlpipeline-minio-artifact --namespace=kubeflow  -o yaml | \
    sed 's/namespace: kubeflow/namespace: admin/' | \
    kubectl create -f -
```

However, the Experiment Run UI still blocks with #395.

**Comment by DomFleischmann** [comment]
This issue should be fixed with the latest Kubeflow 1.4 release, feel free to reopen this issue if it still occurs.
